servicetests/test_cases/test_jobs_execution.py

```diff
@@ -0,0 +1,117 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0.
+
+import argparse
+import json
+import os
+import sys
+import uuid
+
+import boto3
+
+import run_in_ci
+import ci_iot_thing
+
+
+def main():
+    argument_parser = argparse.ArgumentParser(
+        description="Run Jobs test in CI")
+    argument_parser.add_argument(
+        "--config-file", required=True, help="JSON file providing command-line arguments for a test")
+    argument_parser.add_argument(
+        "--input-uuid", required=False, help="UUID for thing name. UUID will be generated if this option is omit")
+    argument_parser.add_argument(
+        "--region", required=False, default="us-east-1", help="The name of the region to use")
+    parsed_commands = argument_parser.parse_args()
+
+    try:
+        iot_client = boto3.client('iot', region_name=parsed_commands.region)
+        secrets_client = boto3.client("secretsmanager", region_name=parsed_commands.region)
+    except Exception as e:
+        print(f"ERROR: Could not make Boto3 iot-data client. Credentials likely could not be sourced. Exception: {e}",
+              file=sys.stderr)
+        return -1
+
+    input_uuid = parsed_commands.input_uuid if parsed_commands.input_uuid else str(uuid.uuid4())
+
+    thing_name = "ServiceTest_Jobs_" + input_uuid
+    policy_name = secrets_client.get_secret_value(
+        SecretId="ci/JobsServiceClientTest/policy_name")["SecretString"]
+
+    # Temporary certificate/key file path.
+    certificate_path = os.path.join(os.getcwd(), "tests/JobsExecution/certificate.pem.crt")
+    key_path = os.path.join(os.getcwd(), "tests/JobsExecution/private.pem.key")
+
+    try:
+        ci_iot_thing.create_iot_thing(
+            thing_name=thing_name,
+            thing_group="CI_ServiceClient_Thing_Group",
+            region=parsed_commands.region,
+            policy_name=policy_name,
+            certificate_path=certificate_path,
+            key_path=key_path)
+    except Exception as e:
+        print(f"ERROR: Failed to create IoT thing: {e}")
+        sys.exit(-1)
+
+    thing_job = 'ERROR' 
+    i = 0;
+    while 'ERROR' in thing_job and i <= 3:
+        try:
+            job_id = secrets_client.get_secret_value(SecretId="ci/JobsServiceClientTest/job_id")["SecretString"]
+            thing_job = iot_client.describe_job_execution(jobId=job_id, thingName=thing_name)
+            print('thing job is {thing_job}');
+            if 'ERROR' in thing_job:
+               i = i + 1;
```

Comment:
Shouldn't we increment `i` in the Exception block also? Though, the limit should be increased.
Also, maybe it's worth adding a 1-second sleep after each unsuccessful attempt, to not bombard server with requests.